py_trees/meta.py

```diff
@@ -188,6 +188,34 @@ def tip(self):
             else:
                 return super(Imposter, self).tip()
 
+        def add_child(self, child):
+            """
+            Adds a child. This function overrides :meth:`~py_trees.composites.Composite.add_child`
+            with additional adoption of child from the original composite.
+            If the original is :class:`~py_trees.behaviour.Behaviour`,
+            this would raise exception as it would do usually.
```

Comment:
At the very least I'd docstring what it actually raises - I suspect `AttributeError` because a non-composite behaviour has no `add_child` method. e.g.

```
Raises:
        AttributeError: if the behaviour is not a composite and thus not permitted to have children.
```
You could probably throw a more meaningful exception than that though. 

Adding the method permanently however is mis-representative of a non-composite behaviour  (i.e. Behaviour does not actually have that method at all). I'd prefer to monkeypatch it in the constructor instead. i.e. check the class to see if it is an instance of composite, and then monkeypatch `add_child`/`add_children` from the original if and only if the parent class instance was composite. How does this sound?